### DISCUSSION TITLE
Access/Modify simulation state when using Orbit BaseEnv

### DISCUSSION THREAD

**Discussion by JChunX** [isaac-sim/IsaacLab#214] [Q&A]
I am trying to port a [MPPI controller](https://github.com/tud-airlab/mppi-isaac) from Isaacgym to Orbit. The controller itself relies on some pretty fine-grained control over environment states for parallelized rollouts. eg. using `set_actor_root_state_tensor`

I know that I can do something similar in Orbit by using the `Scene` APIs, but I would also want to take advantage of the modularity and randomization capabilities of `BaseEnv`. Reading the documentation, it looks like my best option would be to add terms to `ActionsCfg` that directly control joint positions and velocities for the robot, but I don't know if this will also work for all dynamic prims in the scene. Any suggestions would be awesome!

**Comment by Mayankm96** [comment]
Hi @JChunX ,

I think I'd need to understand this a bit more before recommending a path.

In your case, do you want to use the environment for performing rollouts for your controller? In that case, your MPPI controller will wrap around the environment and call the step as needed. I don't think you would implement the MPPI algorithm as an action term itself since the action space should match the space of your controller's output.
